File: python/030.py
# 30) [DESAFIO] Refaça o algoritmo 25, acrescentando o recurso de mostrar que tipo
# de triângulo será formado:
# - EQUILÁTERO: todos os lados iguais
# - ISÓSCELES: dois lados iguais
# - ESCALENO: todos os lados diferentes

segmento_1 = int(input("Digite o primeiro segmento (AB): "))
segmento_2 = int(input("Digite o segundo segmento (CD): "))
segmento_3 = int(input("Digite o terceiro segmento (EF): "))

ab_maior = segmento_1 < (segmento_2 + segmento_3)
cd_maior = segmento_2 < (segmento_1 + segmento_3)
ef_maior = segmento_3 < (segmento_1 + segmento_2)

is_triangulo = ab_maior and cd_maior and ef_maior

is_equilatero = segmento_1 == segmento_2 and segmento_1 == segmento_3
is_isosceles = segmento_1 == segmento_2 or segmento_1 == segmento_3
is_escaleno = segmento_1 != segmento_2 and segmento_1 != segmento_3

if is_triangulo:
    if is_equilatero:
        print("Com os segmentos que você informou foi possível fazer um triângulo EQUILÁTERO: todos os lados iguais")
    elif is_isosceles:
        print("Com os segmentos que você informou foi possível fazer um triângulo ISÓSCELES: dois lados iguais")
    elif is_escaleno:
        print("Com os segmentos que você informou foi possível fazer um triângulo ESCALENO: todos os lados diferentes")
else:
    print("Infelizmente os segmentos que você informou não é possível formar um triângulo!")

File: javascript/030.js
/*
30) [DESAFIO] Refaça o algoritmo 25, acrescentando o recurso de mostrar que tipo
de triângulo será formado:
- EQUILÁTERO: todos os lados iguais
- ISÓSCELES: dois lados iguais
- ESCALENO: todos os lados diferentes
*/

const equalItems = (arr) => {
  let equals = [];
  if (arr instanceof Array) {
    for (let i = 0; i < arr.length; i++) {
      const arrayClone = [...arr];
      arrayClone.splice(i, 1);
      if (arrayClone.includes(arr[i])) {
        equals.push(arr[i]);
      }
    }
  }
  return equals;
};

const isTriangle = (a, b, c) => {
  const l1 = a + b > c;
  const l2 = c + a > b;
  const l3 = b + c > a;
  const equals = equalItems([a, b, c]);
  if (l1 && l2 && l3) {
    switch (equals.length) {
      case 0:
        return "ESCALENO: todos os lados diferentes";
      case 2:
        return "ISÓSCELES: dois lados iguais";
      case 3:
        return "EQUILÁTERO: todos os lados iguais";
    }
  } else return false;
};

console.log(isTriangle(12, 12, 10));
console.log(isTriangle(10, 10, 10));
console.log(isTriangle(5, 2.5, 3.5));
console.log(isTriangle(5, 10, 3.5));
